target/linux/ath79/files/drivers/net/ethernet/atheros/ag71xx/ag71xx_main.c

```diff
@@ -880,17 +882,25 @@ __ag71xx_link_adjust(struct ag71xx *ag, bool update)
 	    !of_device_is_compatible(np, "qca,ar7100-eth"))
 		ag71xx_fast_reset(ag);
 
+	duplex = ag-> duplex;
+	if (ag->builtin_switch)
+		duplex = 1;
+
 	cfg2 = ag71xx_rr(ag, AG71XX_REG_MAC_CFG2);
 	cfg2 &= ~(MAC_CFG2_IF_1000 | MAC_CFG2_IF_10_100 | MAC_CFG2_FDX);
-	cfg2 |= (ag->duplex) ? MAC_CFG2_FDX : 0;
+	cfg2 |= (duplex) ? MAC_CFG2_FDX : 0;
```

Comment:
is there a reason you redefined the pointer to duplex?

I don't see "duplex" by itself anywhere else in the file, just keep it as a pointer

Reply:
I see the typedef of the function but the variable isnt used before this commit, so I would actually just remove the define of the duplex variable

Reply:
This is so I can override it only locally, without affecting the output.